src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -134,6 +140,20 @@
     return null;
   };
 
+  const goToMoveChat = (channel: ChannelList) => {
+    setChat({
+      ...chat,
+      selectedChannel: channel
+    });
+    setChannelAsRead(channel);
+    if (channel?.channelType === ANON_PM) {
+      setSelectedChannelKey(1);
+      return openChat('SampleChatScreen');
+    }
+    setSelectedChannelKey(0);
+    return openChat('SignedChatScreen');
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `goToMoveChat` function sets the selected channel and navigates to the appropriate chat screen based on the channel type. It also sets the channel as read. However, it doesn't handle potential errors that might occur during these operations. Consider adding error handling to improve robustness.
```diff
-    setChat({
-      ...chat,
-      selectedChannel: channel
-    });
-    setChannelAsRead(channel);
-    if (channel?.channelType === ANON_PM) {
-      setSelectedChannelKey(1);
-      return openChat('SampleChatScreen');
-    }
-    setSelectedChannelKey(0);
-    return openChat('SignedChatScreen');
+    try {
+      setChat({
+        ...chat,
+        selectedChannel: channel
+      });
+      await setChannelAsRead(channel);
+      if (channel?.channelType === ANON_PM) {
+        setSelectedChannelKey(1);
+        return openChat('SampleChatScreen');
+      }
+      setSelectedChannelKey(0);
+      return openChat('SignedChatScreen');
+    } catch (error) {
+      console.error("Error moving chat:", error);
+    }

```



<!-- This is an auto-generated comment by OSS CodeRabbit -->